### PR TITLE
chore(docs): Typesafe `metadata` object

### PR DESCRIPTION
**Pull request by carlos-menezes**
Adds missing `Metadata` type to a few declarations.

### CODE REVIEW

**Review by timneutkens — DISMISSED** [review]
All of these are `.js` files 🤔

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 1]
```diff
@@ -179,7 +179,7 @@ export const metadata: Metadata = {
 ```jsx filename="app/layout.js" switcher
 import { Metadata } from 'next'
```
Comment (styfle) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:180]:
```suggestion
```

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 2]
```diff
@@ -193,15 +193,15 @@ export const metadata: Metadata = {
 `title.default` can be used to provide a **fallback title** to child route segments that don't define a `title`.
 
 ```tsx filename="app/layout.tsx"
-export const metadata = {
+export const metadata: Metadata = {
```
Comment (styfle) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:196]:
```suggestion
import { Metadata } from 'next'

export const metadata: Metadata = {
```

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 3]
```diff
@@ -193,15 +193,15 @@ export const metadata: Metadata = {
...
   title: {
     default: 'Acme',
   },
 }
 ```
 
 ```tsx filename="app/about/page.tsx"
-export const metadata = {}
+export const metadata: Metadata = {}
```
Comment (styfle) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:204]:
```suggestion
import { Metadata } from 'next'

export const metadata: Metadata = {}
```

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 4]
```diff
@@ -245,7 +245,7 @@ export const metadata: Metadata = {
 ```jsx filename="app/about/page.js" switcher
 import { Metadata } from 'next'
```
Comment (styfle) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:246]:
```suggestion
```

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 5]
```diff
@@ -520,7 +520,7 @@ export const metadata = {
 ### `robots`
 
 ```tsx
-export const metadata = {
+export const metadata: Metadata = {
```
Comment (styfle) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:523]:
```suggestion
import { Metadata } from 'next'

export const metadata: Metadata = {
```

**Review by delbaoliveira — DISMISSED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 508d5a4d6613cd57da22c3b0f77528b76a1a068e

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by styfle — DISMISSED** [review]
Thanks!

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 6]
```diff
@@ -520,7 +522,9 @@ export const metadata = {
 ### `robots`
 
 ```tsx
-export const metadata = {
+import { Metadata } from 'next'
```
Comment (styfle) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:525]:
```suggestion
import type { Metadata } from 'next'
```

**Review by styfle — DISMISSED** [review]

File: docs/02-app/02-api-reference/04-functions/generate-metadata.mdx [hunk 7]
```diff
@@ -222,9 +225,9 @@ export const metadata: Metadata = {
 ```
 
 ```jsx filename="app/layout.js" switcher
-import { Metadata } from 'next'
+import type { Metadata } from 'next'
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/04-functions/generate-metadata.mdx:228]:
```suggestion
```

**Review by styfle — DISMISSED** [review]

**Review by leerob — DISMISSED** [review]

**Pull request merged**